cirq/contrib/qasm_import/_parser_test.py

```diff
@@ -151,3 +154,106 @@ def test_syntax_error():
 
     with pytest.raises(QasmException, match=r"""Syntax error: 'bla'.*"""):
         parser.parse(qasm)
+
+
+def test_CX_gate():
+    qasm = """
+     OPENQASM 2.0;          
+     qreg q1[2];
+     qreg q2[2];
+     CX q1[0], q1[1];
+     CX q1, q2[0];
+     CX q2, q1;      
+"""
+    parser = QasmParser()
+
+    q1_0 = cirq.NamedQubit('q1_0')
+    q1_1 = cirq.NamedQubit('q1_1')
+    q2_0 = cirq.NamedQubit('q2_0')
+    q2_1 = cirq.NamedQubit('q2_1')
+
+    expected_circuit = Circuit()
+    # CX q1[0], q1[1];
+    expected_circuit.append(cirq.CNOT(q1_0, q1_1))
+    # CX q1, q2[0];
+    expected_circuit.append(cirq.CNOT(q1_0, q2_0))
+    expected_circuit.append(cirq.CNOT(q1_1, q2_0))
+    # CX q2, q1;
+    expected_circuit.append(cirq.CNOT(q2_0, q1_0))
+    expected_circuit.append(cirq.CNOT(q2_1, q1_1))
+
+    parsed_qasm = parser.parse(qasm)
+
+    assert parsed_qasm.supportedFormat is True
+    assert parsed_qasm.qelib1Include is False
+
+    ct.assert_same_circuits(parsed_qasm.circuit, expected_circuit)
+    assert parsed_qasm.qregs == {'q1': 2, 'q2': 2}
+
+
+def test_CX_gate_not_enough_args():
+    qasm = """
+     OPENQASM 2.0;          
+     qreg q[2];
+     CX q[0];
+"""
+    parser = QasmParser()
+
+    with pytest.raises(QasmException,
+                       match=(r"CX only takes 2 arg\(s\) "
+                              r"\(qubits and/or registers\)"
+                              r", got: 1, at line 4")):
+        parser.parse(qasm)
+
+
+def test_cx_gate_mismatched_registers():
+    qasm = """
+     OPENQASM 2.0;
+     qreg q1[2];
```

Comment:
Add a test of an out-of-bounds access raising an error. Try too-large and too-small (negative).

Reply:
👍